helpers/logging_iterator_wrapper.js

```diff
@@ -22,7 +22,13 @@ module.exports = (prefix, valueByKey, iteratee) => {
     numberOfKeysDone += 1;
 
     if (Date.now() - lastLogAt > interval && process.env.TEST === undefined) {
-      const percentageDone = (numberOfKeysDone / valueByKey.size) * 100;
+      let percentageDone;
+      if (valueByKey instanceof Array) {
+        percentageDone = (numberOfKeysDone / valueByKey.length) * 100;
+      } else {
+        percentageDone = (numberOfKeysDone / valueByKey.size) * 100;
```

Comment:
We apply the same "function" on each branch of the logic. It's an opportunity to refactor.

```
const numberOfKeys = valueByKey instanceof Array ? valueByKey.length : valueByKey.size
percentageDone = (numberOfKeysDone / numberOfKeys) * 100;
```

Reply:
Updated